### PR TITLE
Improved use of GitHub Actions + pandas warning fix

### LINKED ISSUE #100: dtype incompatible with float64 warning when writing to pandas df
Windows 10, Python 3.12, PowerShell. I get the following warnings:

```
C:\Users\XXXXXXX\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\ena_upload\ena_upload.py:950: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise in a future error of pandas. Value '['c9900de000f19e3bdebe46b934540449' '1e2ed1f5c2bc84be9a27afec8caf55c1']' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.
  df.loc[:, 'file_checksum'] = md5
No valid checksums found, generate now... done.
```

```
C:\Users\XXXXXX\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\ena_upload\ena_upload.py:982: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value 'My Speices' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.
  df.loc[index, 'scientific_name'] = scientificName
Taxon IDs and scientific names are retrieved
```

```
C:\Users\XXXXXXXX\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\ena_upload\ena_upload.py:586: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value 'ACCESSION' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.
  dataframe.loc[index, 'accession'] = update.loc[index, 'accession']
```

### PR DESCRIPTION
**Pull request by bedroesb**
- [x] Using GH Actions yo test for real by doing a draft submission, and a installation
- [x] Automatic testing for python versions: 3.8, 3.9, 3.10, 3.11, 3.12, 3.13 and following OS versions: ubuntu-latest, macos-latest, windows-latest
- [x] Add fix for pandas df type warning (closes #100) 
- [x] Security fix for pyYAML (related to #97)
- [x] Add/Fix Python 3.13 support
- [x] Drop Python 3.7 support due to incompatibility issues
- [x] Test on Multiple machines a real submission

### CODE REVIEW

**Comment by bedroesb** [comment]
Sorry I just noticed I still have to test some local changes I didn't push, sorry for the spam!

**Comment by bedroesb** [comment]
@bgruening I don't understand why I didn't add all these test before, but at least they are now there ... :)

File: .github/workflows/python-app.yml [hunk 1]
```diff
@@ -10,25 +10,38 @@ on:
...
 jobs:
+  
   build:
-
-    runs-on: ubuntu-latest
-
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        version: ["3.8", "3.9", "3.10", "3.11", "3.12", "3.13"]
```
Comment (bgruening) [inline .github/workflows/python-app.yml:18]: I guess it would be enough if we test for the lower and upper bound?

File: ena_upload/ena_upload.py
Comment (bgruening) [inline ena_upload/ena_upload.py:1]: This file is hard to review, I guess we trust the tests :)

**Review by bgruening — APPROVED** [review]
Coool! Thanks @bedroesb 

Only one small comment.

File: .github/workflows/python-app.yml (hunk 1 above)
Comment (bedroesb) [reply .github/workflows/python-app.yml]: genius!

File: ena_upload/ena_upload.py
Comment (bedroesb) [reply ena_upload/ena_upload.py:1]: I wonder what went wrong with this file :thinking: I added multiple `.Series(dtype='str')` to keep Pandas happy

**Comment by bgruening** [comment]
Cool, great stuff @bedroesb

**Pull request merged**